### PR TITLE
Fix PR subscription error "You cannot update an existing Post"

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR fixes an issue in the pull request event webhook logic where we reuse the same post struct for multiple calls to `CreatePost`, resulting in this error:

`CreatePost: You cannot update an existing Post.`

This PR applies a fix to make it so we instantiate a `Post` struct in the subscriptions loop, and only when we are going to create a post for that given subscription for the most part.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-github/issues/754

### CODE REVIEW

File: server/plugin/webhook.go [hunk 1]
```diff
@@ -829,6 +817,18 @@ func (p *Plugin) postIssueCommentEvent(event *github.IssueCommentEvent) {
 			continue
 		}
 
+		post := &model.Post{
+			UserId: p.BotUserID,
+			Type:   "custom_git_comment",
+		}
+
+		repoName := strings.ToLower(repo.GetFullName())
+		commentID := event.GetComment().GetID()
```
Comment (ayusht2810) [inline server/plugin/webhook.go:826]: I think we can move this out of the loop

File: server/plugin/webhook.go [hunk 2]
```diff
@@ -975,6 +962,19 @@ func (p *Plugin) postPullRequestReviewCommentEvent(event *github.PullRequestRevi
...
 		}
 
+		post := &model.Post{
+			UserId:  p.BotUserID,
+			Type:    "custom_git_pull_review_comment",
+			Message: newReviewMessage,
+		}
+
+		repoName := strings.ToLower(repo.GetFullName())
+		commentID := event.GetComment().GetID()
```
Comment (ayusht2810) [inline server/plugin/webhook.go:972]: same comment as above

File: server/plugin/webhook.go [hunk 3]
```diff
@@ -409,17 +404,22 @@ func (p *Plugin) postPullRequestEvent(event *github.PullRequestEvent) {
...
+			continue
+		}
+
 		repoName := strings.ToLower(repo.GetFullName())
 		prNumber := event.GetPullRequest().Number
 
+		post := &model.Post{
+			UserId: p.BotUserID,
+			Type:   "custom_git_pr",
+		}
```
Comment (hanzei) [inline server/plugin/webhook.go:417]: Should we move the post creation into a separate method and reuse it?

File: server/plugin/webhook.go (hunk 1 above)
Comment (hanzei) [reply server/plugin/webhook.go:826]: 1/5 that getting the commentID and assigning it as a prop should stay together as they semantically belong together.

File: server/plugin/webhook.go
Comment (hanzei) [inline server/plugin/webhook.go:1]: I would love to see tests for these changes. One can only dream...

**Review by hanzei — CHANGES_REQUESTED** [review]
Should the code in `postDeleteEvent` also get updated? It seems to suffer from the same issue.

**Review by hanzei — DISMISSED** [review]
Should the code in `postDeleteEvent` also get updated? It seems to suffer from the same issue.

File: server/plugin/webhook.go
Comment (hanzei) [reply server/plugin/webhook.go:1]: No, the tests are not a requirement. 

The lack of tests is one of the reasons that heavy manual QA testing is required.

File: server/plugin/webhook.go (hunk 1 above)
Comment (mickmister) [reply server/plugin/webhook.go:826]: @ayusht2810 Note that these methods being called are not performing network requests. They are essentially helper methods that are accessing fields on the struct for us. @hanzei Sure we can keep it as is. `strings.ToLower` is happening each time but that's the only concern I could potentially have. The repo name is always going to be short.

**Review by Kshitij-Katiyar — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
@mickmister Can we please fix the lint errors here ?

File: server/plugin/webhook.go (hunk 1 above)
Comment (ayusht2810) [reply server/plugin/webhook.go:803]: @mickmister Sure, that's fine with me.

**Review by ayusht2810 — APPROVED** [review]

**Comment by mickmister** [comment]
@AayushChaudhary0001 Can you take a look at this PR when you have the chance? Essentially all webhook events need to be tested

**Review by AayushChaudhary0001 — APPROVED** [review]
This PR has been tested for all the webhook events, and all of them were found working fine, LGTM. Approved.

**Review by fmartingr — APPROVED** [review]
👌

**Pull request merged**